src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1290,7 +1311,22 @@ std::unique_ptr<StorageTargetRHandle> SQLStorage::openTargetFile(const Uptane::T
 void SQLStorage::removeTargetFile(const std::string& filename) {
   SQLite3Guard db = dbConnection();
 
-  auto statement = db.prepareStatement<std::string>("DELETE FROM target_images WHERE filename=?;", filename);
```

Comment:
Should the two operations belong in one transaction?

Reply:
Yep, I will add transaction